src/tests/io/test_open.f90

```diff
@@ -38,21 +47,25 @@ program test_open
 filename = get_outpath() // "/io_open.stream"
 
 ! Test mode "w"
-u = open(filename, "wb")
+u = open(filename, "wb", io)
+call assert(io == 0)
```

Comment:
I would leave these tests without `io` present, thus letting the built-in `open` to generate an error.

The reason for that is that this is how mostly the `open` function will be used. So that's what needs to be tested the most.

In addition, let's add tests where `io` is also tested (both zero and non-zero). These tests (at least initially) do not have to be so thorough.